enhancements/microshift/microshift-telemetry.md

```diff
@@ -0,0 +1,293 @@
+Using defaults (1 payload every 24h, with 1h samples for all dynamic metrics)
+would be under 40 KB.
+
+### Sensitive data
+There is no user or private data in any of the metrics in MicroShift reports.
+
+### Opting out
+MicroShift may not always have the possibility of sending metrics. There may be
+disconnected clusters, constrained environments where external traffic could be
+audited, customers who simply do not want to share this information, etc. For
+these reasons there must be an opt-out for this functionality, as OpenShift
+already [allows](https://docs.openshift.com/container-platform/4.17/support/remote_health_monitoring/opting-out-of-remote-health-reporting.html).
+
+Taking advantage of the configuration options that the feature requires, an
+enable/disable toggle is provided.
+
+### Workflow Description
+**MicroShift** is the MicroShift main process.
+
+1. MicroShift starts up.
+2. MicroShift reads configuration. If telemetry is not enabled, finish here. If telemetry is enabled proceed to next step.
+3. Collect all metrics and send them. Include dynamic metrics from sampling if available. Retry if failed.
+4. Schedule next send for `telemetry.reportInterval`.
+5. Every `telemetry.sampleInterval` collect all dynamic metrics and store them in memory.
+6. Wait until `telemetry.reportInterval` and go to step 3.
+
+```mermaid
+sequenceDiagram
+    participant MicroShift
+    participant Red Hat Telemetry
+    MicroShift ->> MicroShift: Start up. Read configuration
+    loop Send Report
+    MicroShift ->> MicroShift: Collect static and dynamic metrics
+    loop Retries
+        MicroShift -->> Red Hat Telemetry: Send WriteRequest
+        Red Hat Telemetry ->> MicroShift: 200 Ok
+    end
+    MicroShift ->> MicroShift: Schedule next send loop to telemetry.reportInterval
+    loop Sampling
+    MicroShift ->> MicroShift: Collect dynamic metrics every telemetry.samplingInterval
+    MicroShift ->> MicroShift: Store dynamic metrics until next report
+    end
+    end
+    
+```
+
+### API Extensions
+As described above, the feature needs to be configurable as there could be different reasons why MicroShift admins would not want their clusters to connect to Red Hat telemetry.
+The following changes in the configuration file are proposed:
+```yaml
+telemetry:
+  status: <Enabled|Disabled> # Defaults to Enabled
+  sendingInterval: <Duration> # Defaults to 24h
+  samplingInterval: <Duration> # Defaults to 1h
+```
+
+### Topology Considerations
+#### Hypershift / Hosted Control Planes
+N/A
+
+#### Standalone Clusters
+N/A
+
+#### Single-node Deployments or MicroShift
+Enhancement is intended for MicroShift only.
+
+### Implementation Details/Notes/Constraints
+N/A
+
+### Risks and Mitigations
+Using the telemeter API requires connected clusters, and this might not always
+be the case with MicroShift. For those clusters that are not connected there
+is no easy way of getting metrics from them. A possible mitigation would be to
```

Comment:
Or is this something we gather that an admin can then use when an edge device is about to be upgraded or decommissioned?

Reply:
That would also be interesting, it would need to be connected though. Let me give it some thought and include it here.